### PR TITLE
fix: correct Show in Source navigation (#line offset + unquoted filenames)

### PR DESCRIPTION
**Pull request by theodevelop**
## Type of change
- [x] Bug fix
- [ ] New feature
- [ ] Refactor / internal improvement
- [ ] Docs / config only

## What does this PR do?
- Fixed `showInSource` jumping to the wrong line when the cursor is below a `#line` directive — now applies the correct offset `sourceLine + (cursorLine − directiveLine)`
- Fixed `#line` detection not matching unquoted filenames (`#line 28 parser.y` without quotes)
- Extracted pure helpers to `client/src/lineDirectiveUtils.ts` for testability (no VS Code mock needed)

## Closes

Closes #29 #27 (comment by @GitMensch)

## Changes

- `client/src/lineDirectiveUtils.ts` — new module: `isGeneratedFile`, `LineDirective`, `findNearestLineDirective`
- `client/src/lineDirectiveNavigation.ts` — imports from utils, applies offset formula
- `tests/test-line-directive.ts` — 26 assertions (quoted, unquoted, edge cases)

## How to test manually
- [x] `TS_NODE_PROJECT=tsconfig.base.json npx ts-node tests/test-line-directive.ts` → 26 passed, 0 failed
- [x] `npm run compile` → no errors
- [ ] Manual: cursor N lines below `#line M` → navigates to line `M + N` in source
- [ ] Manual: unquoted `#line N filename.y` → directive detected and navigation works
- [ ] Manual: cursor above all `#line` directives → warning message shown

## Checklist
- [x] `npm run compile` passes with no new errors
- [x] Tests added or updated (`npx ts-node ...`)
- [x] Manual test done in VS Code
- [ ] CHANGELOG.md updated
- [x] No unintended files staged (node_modules, .env, dist...)

### CODE REVIEW

File: client/src/lineDirectiveNavigation.ts [hunk 1]
```diff
@@ -70,7 +44,8 @@ export async function showInSource(): Promise<void> {
     sourcePath = found;
   }
 
-  const targetLine = Math.max(0, directive.sourceLine - 1); // convert to 0-based
+  const offset = cursorLine - directive.directiveLine; // lines between directive and cursor
```
Comment (GitMensch) [inline client/src/lineDirectiveNavigation.ts:47]: Is that correct? I think yes, but the later _use_ of the offset should possibly have another `+ 1`:

Image

goes to

Image


note that `#line 6287 "../../cobc/parser.y"` says that _the next line_ is line 6287;

cursor line (6290) - directive line (6287) = 3 (+1 --> 4 if considered from the #line itself), but the position is only 3 lines below the #line .


Apart from that the feature works fine.
